types/observer-mixin.d.ts

```diff
@@ -36,7 +50,7 @@ export interface ObserverMixin {
    * `this` set to the model constructor, e.g. `User`.
    * @end
    */
-  observe(operation: string, listener: Listener): void;
+  observe(operation: string, listener: Listener<OperationHookContext<PersistedModelClass>>): void;
```

Comment:
It makes me wonder, can we use `this` as the value of the generic argument?

```suggestion
  observe(operation: string, listener: Listener<OperationHookContext<this>>): void;
```

If yes, then I think such typing will express exactly what we need (see my other comment https://github.com/strongloop/loopback-datasource-juggler/pull/1820#pullrequestreview-365481860 for more context):
> if a class T extends Model, then it will include ObserverMixin<T> methods

Reply:
ok, so it's not possible to use `this` in the context of a static method, but there is a workaround described in https://github.com/microsoft/TypeScript/issues/5863#issuecomment-410887254.

```ts
observe<T>(this: T, operation: string, listener: Listener<OperationHookContext<T>>): void;
```

Here a standalone example I used to verify this approach, it compiles cleanly for me.

```ts
interface Context<T> {
  target: T;
}

type Listener<T> = (ctx: Context<T>) => void;

class Base {
  static observe<T extends Base>(this: T, listener: Listener<T>): void {
    listener({target: this});
  }
}

class Foo extends Base {
  static isFoo = true;
}

Foo.observe(ctx => {
  console.log('target is foo', ctx.target.isFoo);
});
```

Anyhow, I think this part may be out of scope of this pull request, I have some ideas how to address it in https://github.com/strongloop/loopback-datasource-juggler/pull/1823.